parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToTypeWithoutAssigningIds.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.parquet;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type.Repetition;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+/**
+ * A visitor that converts a {@link MessageType} to a {@link Type} in Iceberg without assigning ids.
+ * Columns without ids are pruned.
+ */
+public class MessageTypeToTypeWithoutAssigningIds extends BaseMessageTypeToType {
+  private static final Joiner DOT = Joiner.on(".");
+
+  private final Map<String, Integer> aliasToId = Maps.newHashMap();
+
+  MessageTypeToTypeWithoutAssigningIds() {}
+
+  public Map<String, Integer> getAliases() {
+    return aliasToId;
+  }
+
+  @Override
+  public Type message(MessageType message, List<Type> fields) {
```

Comment:
This is the same in both visitors. Should it be moved into the base class?

Reply:
Updated `MessageTypeToType` instead.